server/src/main/java/org/elasticsearch/health/HealthService.java

```diff
@@ -80,30 +98,96 @@ public List<HealthIndicatorResult> getHealth(@Nullable String indicatorName, boo
+                getHealthNoHealthInfo(listener, indicatorName, preflightResults, filteredIndicators.toList(), explain);
+            }
         } else {
             // Mark remaining indicators as UNKNOWN
             HealthIndicatorDetails unknownDetails = healthUnknownReason(preflightResults, explain);
-            filteredIndicatorResults = filteredIndicators.map(
+            List<HealthIndicatorResult> filteredIndicatorResults = filteredIndicators.map(
                 service -> generateUnknownResult(service, UNKNOWN_RESULT_SUMMARY_PREFLIGHT_FAILED, unknownDetails)
-            );
+            ).toList();
+            combineResultsAndNotifyListener(listener, indicatorName, preflightResults, filteredIndicatorResults);
         }
+    }
 
+    /**
+     * This method calls calculate on the filteredIndicators using HealthInfo.EMPTY_HEALTH_INFO, and passes on these results and the
+     * preflight results to aggregateHealthAndNotifyListener.
+     * @param listener A listener to be notified of the list of all HealthIndicatorResult if indicatorName is null, or one
+     *                           HealthIndicatorResult if indicatorName is not null
+     * @param indicatorName If not null, the returned results will only have this indicator
+     * @param preflightResults The results of the preflight health indicators
+     * @param filteredIndicators The remaining health indicators that need to be calculated
+     * @param explain       Whether to compute the details portion of the results
+     */
+    private void getHealthNoHealthInfo(
+        ActionListener<List<HealthIndicatorResult>> listener,
+        String indicatorName,
+        List<HealthIndicatorResult> preflightResults,
+        List<HealthIndicatorService> filteredIndicators,
+        boolean explain
+    ) {
+        combineResultsAndNotifyListener(
+            listener,
+            indicatorName,
+            preflightResults,
+            filteredIndicators.stream().map(service -> service.calculate(explain, HealthInfo.EMPTY_HEALTH_INFO)).toList()
+        );
+    }
+
+    /**
+     * This method combines the preflight indicator results with the non-preflight indicator results, and notifies the listener. If there
+     * are no results and the indicator name is not null, the listener will be notified of failure. Otherwise the listener will be
+     * notified with the results.
+     * @param listener A listener to be notified of the list of all HealthIndicatorResult if indicatorName is null, or one
+     *                               HealthIndicatorResult if indicatorName is not null
+     * @param indicatorName If not null, the returned results will only have this indicator
+     * @param preflightResults The results of the preflight health indicators
+     * @param filteredIndicatorResults The results of the non-preflight health indicators
+     */
+    private void combineResultsAndNotifyListener(
+        ActionListener<List<HealthIndicatorResult>> listener,
```

Comment:
I believe for these APIs it's usually that we have the client first (to indicate a remote call), the business logic arguments, and last the listener 

e.g. `public static void closePointInTime(Client client, String pointInTimeId, ActionListener<Boolean> listener)`

Reply:
TIL, makes perfect sense.